runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -293,6 +293,10 @@ public void gc(long trimMark) {
      *                             the supplied version.
      */
     public void rollbackObjectUnsafe(long timestamp) {
+        if (object.getVersionPolicy() == ICorfuVersionPolicy.MONOTONIC) {
```

Comment:
If going back in time is not allowed, how could the upper layer client know the restriction. Is there any exception thrown?

Reply:
Well, when the client creates the SMR object, they also defines the versioning policy. In other words, this option is explicitly defined by the user.